### PR TITLE
PIM-5589: Replace attribute groups YML import by CSV

### PR DESCRIPTION
**Pull request by pierallard**
**Description (for Contributor and Core Developer)**

Remove old YML import for attribute_groups, replace it by a CSV.

**Definition Of Done (for Core Developer only)**

```
| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | y
| Added Behats                      | n
| Changelog updated                 | TODO
| Review and 2 GTM                  | TODO
| Migration script                  | n
| Tech Doc                          | n
```

### CODE REVIEW

File: src/Pim/Bundle/BaseConnectorBundle/Resources/config/processors.yml [hunk 1]
```diff
@@ -97,6 +97,7 @@ services:
         arguments:
             - %pim_catalog.entity.association_type.class%
 
+    # @deprecated Will be removed when all the new import system will be ready
```
Comment (nidup) [inline src/Pim/Bundle/BaseConnectorBundle/Resources/config/processors.yml]: will be removed in 1.6

File: src/Pim/Component/Catalog/Factory/AttributeGroupFactory.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Pim\Component\Catalog\Factory;
```
Comment (nidup) [inline src/Pim/Component/Catalog/Factory/AttributeGroupFactory.php]: missing space, run php-cs-fixer please

File: src/Pim/Component/Connector/ArrayConverter/Flat/AttributeGroupStandardConverter.php [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            $convertedItem['label'][$matches[1]] = $data;
+        } else {
+            $convertedItem[$field] = explode(',', $data);
+        }
+
+        return $convertedItem;
+    }
+
+    /**
+     * TODO: Should be refactored with Pim\Component\Connector\ArrayConverter\Flat\ChannelStandardConverter
```
Comment (nidup) [inline src/Pim/Component/Connector/ArrayConverter/Flat/AttributeGroupStandardConverter.php]: agree, imho all converters could use it, please extract in a new class / service and use it from both

**Comment by nidup** [comment]
extract a new class / service for "validateRequiredFields", add a note in CHANGELOd.md then GTM

File: spec/Pim/Component/Connector/ArrayConverter/FieldsRequirementValidatorSpec.php [hunk 4]
```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace spec\Pim\Component\Connector\ArrayConverter;
+
+class FieldsRequirementValidatorSpec extends \PhpSpec\ObjectBehavior
```
Comment (aRn0D) [inline spec/Pim/Component/Connector/ArrayConverter/FieldsRequirementValidatorSpec.php]: it miss it_is_initializable method

File: src/Pim/Component/Connector/ArrayConverter/Flat/AttributeGroupStandardConverter.php [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+ * @author    Pierre Allard <pierre.allard@akeneo.com>
+ * @copyright 2016 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class AttributeGroupStandardConverter implements StandardArrayConverterInterface
+{
+    /** @var FieldsRequirementValidator */
+    protected $validator;
+
+    public function __construct(FieldsRequirementValidator $validator)
```
Comment (solivier) [inline src/Pim/Component/Connector/ArrayConverter/Flat/AttributeGroupStandardConverter.php]: missing doc

File: src/Pim/Component/Connector/ArrayConverter/FieldsRequirementValidator.php [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+use Pim\Component\Connector\Exception\ArrayConversionException;
+
+/**
+ * Simple validator to check if required fields are present and filled.
+ *
+ * @author    Pierre Allard <pierre.allard@akeneo.com>
+ * @copyright 2016 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class FieldsRequirementValidator
```
Comment (solivier) [inline src/Pim/Component/Connector/ArrayConverter/FieldsRequirementValidator.php]: :+1: cleaner this way

File: src/Pim/Bundle/InstallerBundle/Resources/config/batch_jobs.yml [hunk 7]
```diff
@@ -20,21 +20,21 @@ connector:
                     parameters:
                         batchSize: 1
 
-        fixtures_attribute_group_yml:
-            title: pim_installer.jobs.fixtures_attribute_group_yml.title
+        fixtures_attribute_group_csv:
```
Comment (solivier) [inline src/Pim/Bundle/InstallerBundle/Resources/config/batch_jobs.yml]: we don't move the old job configuration in the deprecated job conf file ?

**Comment by solivier** [comment]
nice PR, GTM after comments fixed

File: spec/Pim/Component/Connector/ArrayConverter/FieldsRequirementValidatorSpec.php (hunk 4 above)
Comment (pierallard) [inline spec/Pim/Component/Connector/ArrayConverter/FieldsRequirementValidatorSpec.php]: Done!

File: src/Pim/Bundle/InstallerBundle/Resources/config/batch_jobs.yml (hunk 7 above)
Comment (pierallard) [inline src/Pim/Bundle/InstallerBundle/Resources/config/batch_jobs.yml]: It's the fixtures installer, we have to remove it. See @nidup 's PR : https://github.com/akeneo/pim-community-dev/pull/4029/files#diff-88d2c1043e00ceb059e822d9b3bd59bfL39

**Pull request merged**